### PR TITLE
Improve metrics for the transaction pool.

### LINKED ISSUE #8084: Better Event Metrics
There are a number of metrics that would be really helpful for understanding event intake and event creation behavior.

### PR DESCRIPTION
**Pull request by cody-littley**
In this PR, I improve the metrics for the transaction pool. While I was at it, I also did some much needed maintenance on the transaction pool object itself (no new features, just cleanup).

Closes #8084

### CODE REVIEW

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java [hunk 1]
```diff
@@ -231,31 +235,25 @@ public synchronized boolean submitTransaction(final ConsensusTransactionImpl tra
...
      * @return the number of transactions
      */
-    public synchronized int getTransEventSize() {
-        return transEvent.size();
+    private synchronized int getBufferedTransactionCount() {
+        return bufferedTransactions.size();
     }
 
     /**
      * get the number of transactions in the priorityTransEvent queue
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java:243]: javadoc still references old member name `priorityTransEvent`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java [hunk 2]
```diff
@@ -231,31 +235,25 @@ public synchronized boolean submitTransaction(final ConsensusTransactionImpl tra
      *
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java:235]: Javadoc still references old member name `transEvent`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    private static final SpeedometerMetric.Config REJECTED_APP_TRANSACTIONS_CONFIG = new SpeedometerMetric.Config(
+                    "platform", "rejectedAppTransactions")
+            .withDescription("Cycled when an app transaction is submitted to the transaction pool and not accepted.");
+    private final SpeedometerMetric rejectedAppTransactions;
+
+    private static final SpeedometerMetric.Config SUBMITTED_PLATFORM_TRANSACTIONS_CONFIG = new SpeedometerMetric.Config(
+                    "platform", "submittedPlatformTransactions")
+            .withDescription(
+                    "Cycled when an app transaction is submitted (platform transactions are always accepted).");
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java:46]: "Cycled when _a platform_ transaction is submitted..."

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    private final SpeedometerMetric submittedPlatformTransactions;
+
+    /**
+     * Create metrics for the transaction pool.
+     *
+     * @param platformContext the platform context
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java:52]: missing param docs

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+        final Metrics metrics = platformContext.getMetrics();
+
+        acceptedAppTransactions = metrics.getOrCreate(ACCEPTED_APP_TRANSACTIONS_CONFIG);
+        rejectedAppTransactions = metrics.getOrCreate(REJECTED_APP_TRANSACTIONS_CONFIG);
+        submittedPlatformTransactions = metrics.getOrCreate(SUBMITTED_PLATFORM_TRANSACTIONS_CONFIG);
+
+        metrics.getOrCreate(new FunctionGauge.Config<>(
+                        PLATFORM_CATEGORY, "bufferedTransactions", Integer.class, getBufferedTransactionCount)
+                .withDescription("transEvent queue size")
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java:67]: description doesn't match updated naming

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                .withUnit("count"));
+        metrics.getOrCreate(new FunctionGauge.Config<>(
+                        PLATFORM_CATEGORY,
+                        "bufferedPriorityTransactions",
+                        Integer.class,
+                        getPriorityBufferedTransactionCount)
+                .withDescription("priorityTransEvent queue size")
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java:74]: description doesn't match updated naming

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SyncManagerTest.java [hunk 7]
```diff
@@ -88,13 +89,13 @@ public SyncManagerTestData(final SwirldStateManager swirldStateManager) {
             startUpEventFrozenManager = mock(StartUpEventFrozenManager.class);
             final Random random = getRandomPrintSeed();
             hashgraph = new DummyHashgraph(random, 0);
-            final TransactionConfig transactionConfig =
-                    new TestConfigBuilder().getOrCreateConfig().getConfigData(TransactionConfig.class);
-            eventTransactionPool = spy(new EventTransactionPool(new NoOpMetrics(), transactionConfig, null));
+            final PlatformContext platformContext =
+                    TestPlatformContextBuilder.create().build();
+
+            eventTransactionPool = spy(new TransactionPool(platformContext, null));
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SyncManagerTest.java:95]: nit: member name could be changed to `transactionPool`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java [hunk 8]
```diff
@@ -39,58 +36,63 @@
  * Store a list of transactions created by self, both system and non-system, for wrapping in the next event to be
  * created.
  */
-public class EventTransactionPool implements TransactionPool, TransactionSupplier {
+public class TransactionPool implements TransactionSupplier, EventCreationRule {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java:39]: nit: Line 218 of `SwirldsPlatform` still mentions "event transaction pool"

**Review by alittley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java:243]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java:235]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java:46]: Good catch, fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java:52]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java (hunk 5 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java:67]: updated

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java (hunk 6 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java:74]: updated

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SyncManagerTest.java (hunk 7 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SyncManagerTest.java:95]: updated

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java (hunk 8 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java:39]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java [hunk 9]
```diff
@@ -64,14 +66,14 @@ public TransactionPoolMetrics(
 
         metrics.getOrCreate(new FunctionGauge.Config<>(
                         PLATFORM_CATEGORY, "bufferedTransactions", Integer.class, getBufferedTransactionCount)
-                .withDescription("transEvent queue size")
+                .withDescription("The number of transactions waiting to be inserted into an event.")
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java:69]: Is it worth specifying that these are "non-priority" transactions?

**Review by alittley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java (hunk 9 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPoolMetrics.java:69]: updated wording

**Review by alittley — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Pull request merged**